pkg/pillar/types/certinfotypes.go

```diff
@@ -4,20 +4,50 @@
 package types
 
 import (
-	"encoding/base64"
+	"encoding/hex"
+	"time"
 	//zcert "github.com/lf-edge/eve/api/go/certs"
 )
 
-// ControllerCertificate : controller certicate
-type ControllerCertificate struct {
-	//	HashAlgo zcert.ZCertHashAlgorithm
-	//	Type     zcert.ZCertType
-	//	Attr     zcert.ZCertProperties
+// ControllerCertConfig : controller certicate
+// config received from controller
+type ControllerCertConfig struct {
+	//HashAlgo zcert.CertHashAlgorithm
+	//Type     zcert.ZCertType
 	Cert     []byte
 	CertHash []byte
 }
 
 // Key :
-func (cert *ControllerCertificate) Key() string {
-	return base64.StdEncoding.EncodeToString(cert.CertHash)
+func (cert *ControllerCertConfig) Key() string {
+	return hex.EncodeToString(cert.CertHash)
+}
+
+// ControllerCertStatus : controller certicate
+// status
+type ControllerCertStatus struct {
+	//HashAlgo zcert.CertHashAlgorithm
+	//Type     zcert.ZCertType
+	Cert     []byte
+	CertHash []byte
+	ErrorInfo
```

Comment:
Is there an API to report this to the controller?

Reply:
A new API will be defined. It will periodically publish the object cipher status and error, if any.

Reply:
Please create a story for that.